optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingGenerator.java

```diff
@@ -446,31 +454,75 @@ private void createTalkList(ConferenceSolution solution, int talkListSize) {
             talk.setUndesiredTimeslotTagSet(new LinkedHashSet<>());
-            talk.setRequiredRoomTagSet(new LinkedHashSet<>());
-            talk.setPreferredRoomTagSet(new LinkedHashSet<>());
-            talk.setProhibitedRoomTagSet(new LinkedHashSet<>());
-            talk.setUndesiredRoomTagSet(new LinkedHashSet<>());
-            talk.setMutuallyExclusiveTalksTagSet(new LinkedHashSet<>());
-            talk.setPrerequisiteTalkSet(new LinkedHashSet<>());
+            talk.setRequiredRoomTagSet(requiredRoomTagSet);
+            talk.setPreferredRoomTagSet(preferredRoomTagSet);
+            talk.setProhibitedRoomTagSet(prohibitedRoomTagSet);
+            talk.setUndesiredRoomTagSet(undesiredRoomTagSet);
+            talk.setMutuallyExclusiveTalksTagSet(mutuallyExclusiveTagSet);
+            talk.setPrerequisiteTalkSet(prerequisiteTalkCodeSet);
             talk.setFavoriteCount(random.nextInt(1000));
-            talk.setCrowdControlRisk(0); // Disabled for now: the unsolved schedules must have a feasible solution
+            if (random.nextDouble() < 0.02) { // Need an even number of talks with crowd control > 1 for a feasible solution
+                talk.setCrowdControlRisk(1);
+                Talk pairedTalk = talkList.get(random.nextInt(talkList.size()));
+                while (pairedTalk.getCrowdControlRisk() != 0 || !pairedTalk.getTalkType().equals(talk.getTalkType())) {
+                    pairedTalk = talkList.get(random.nextInt(talkList.size()));
+                }
+                pairedTalk.setCrowdControlRisk(1);
+            }
             logger.trace("Created talk with code ({}), title ({}) and speakers ({}).",
                     talk.getCode(), talk.getTitle(), speakerList);
             talkList.add(talk);
         }
+
         Talk pinnedTalk = talkList.get(labTalkCount + random.nextInt(talkListSize - labTalkCount));
         pinnedTalk.setPinnedByUser(true);
         pinnedTalk.setTimeslot(solution.getTimeslotList().stream()
                 .filter(timeslot -> timeslot.getTalkTypeSet().contains(breakoutTalkType)).findFirst().get());
         pinnedTalk.setRoom(solution.getRoomList().get(0));
+
+        Talk publishedTalk = talkList.get(random.nextInt(labTalkCount));
+        publishedTalk.setTimeslot(solution.getTimeslotList().stream()
+                .filter(timeslot -> timeslot.getTalkTypeSet().contains(labTalkType)).findFirst().get());
+        publishedTalk.setRoom(solution.getRoomList().stream()
+                .filter(room -> room.getTalkTypeSet().contains(labTalkType)).findFirst().get()); // In this data set rooms has only one talkType
+        publishedTalk.setPublishedTimeslot(publishedTalk.getTimeslot());
+        publishedTalk.setPublishedRoom(publishedTalk.getRoom());
+
         solution.setTalkList(talkList);
     }
 
+    private void setRoomTags(Set<String> requiredRoomTagSet, Set<String> preferredRoomTagSet, Set<String> prohibitedRoomTagSet, Set<String> undesiredRoomTagSet) {
```

Comment:
this method should not start with the prefix "set" as that applies it's changing something on this class.
Let's call it initializeRoomTagSets(...)?